pkg/endpoint/id/id.go

```diff
@@ -70,13 +70,16 @@ func NewID(prefix PrefixType, id string) string {
 	return string(prefix) + ":" + id
 }
 
-// NewIPPrefixID returns an identifier based on the IP address specified
-func NewIPPrefixID(ip net.IP) string {
-	if ip.To4() != nil {
-		return NewID(IPv4Prefix, ip.String())
+// NewIPPrefixID returns an identifier based on the IP address specified. An empty
+// string is returned if ip is nil.
+func NewIPPrefixID(ip *netip.Addr) string {
```

Comment:
There's no need to make this accept a pointer type to check for nil because there's an `IsValid()` method for `netip.Addr`

https://pkg.go.dev/net/netip#Addr.IsValid